### PR TITLE
Add autofill for posthog props in plugin config

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

This autofills two PostHog-related config values for plugins:

- `posthogHost`
- `posthogApiKey`

Ideally in the future we would expand the plugin server with more PostHog functionality beyond events. However, at the moment there are things that can be done by hitting the PostHog API. For example, the [release tracking plugins](https://github.com/PostHog/plugin-snippets/tree/master/gitlab-release-tracker-plugin) hit the annotations endpoint to generate annotations for each new release in your repo. 

To do that, they need a host and a key. So this PR grabs the host from the window and creates an API key so that you don't have to.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/scenes/plugins/pluginsLogic.ts [hunk 1]
```diff
@@ -158,6 +159,17 @@ export const pluginsLogic = kea<
...
+        personalApiKey: [
+            [] as PersonalAPIKeyType[],
+            {
+                createKey: async (label: string) => {
+                    const newKey: PersonalAPIKeyType = await api.create('api/personal_api_keys/', {
+                        label,
+                    })
+                    return newKey
+                },
+            },
```
Comment (mariusandra) [inline frontend/src/scenes/plugins/pluginsLogic.ts:171]: The default value is an `[]` array of type `PersonalAPIKeyType[]`, yet the loader returns a single `PersonalAPIKeyType`

File: frontend/src/types.ts [hunk 2]
```diff
@@ -311,3 +311,12 @@ export interface PluginErrorType {
...
 }
+
+export interface PluginConfigSchemaType {
+    hint: string
+    key: string
+    name: string
+    order: number
+    required: boolean
+    type: string
+}
```
Comment (mariusandra) [inline frontend/src/types.ts:322]: This is a shared type that you can just get with

```
import { PluginConfigSchema } from '@posthog/plugin-scaffold
```

... so we shouldn't re-define it here

File: frontend/src/scenes/plugins/PluginDrawer.tsx [hunk 3]
```diff
@@ -31,22 +32,39 @@ function EnabledDisabledSwitch({
...
+                    if (!personalApiKey || personalApiKey.length === 0) {
+                        createKey('Plugins')
+                    } else {
+                        pluginConfig.posthogApiKey = personalApiKey.value
+                    }
+                }
+                if (pluginConfigSchemaKeys.includes('posthogHost') && !pluginConfig.posthogHost) {
+                    pluginConfig.posthogHost = window.location.origin
+                }
+            }
```
Comment (mariusandra) [inline frontend/src/scenes/plugins/PluginDrawer.tsx:59]: I'd move all of this into a logic. That's where `editingPlugin.pluginConfig` comes from after all...

**Review by mariusandra — CHANGES_REQUESTED** [review]
Hey, some thoughts below regarding this implementation. 

I understand this is needed for some plugins that want to talk to the API (e.g. the github annotations plugin). This will work, though a nicer solution would be to autogenerate and pre-fill this stuff in the plugin-server... and then expose some `api` object that can be called that has the right authentication already pre-filled. That's obviously more work though... :)

**Comment by yakkomajuri** [comment]
Finally had a little time to come back to this - all addressed @mariusandra 

Also completely agree it is far from the best solution. I'd love to have access to more if not all of the PostHog API directly in the plugin, but while that's not the case this will just make things easier for people looking to use the GitHub and GitLab plugins (which work on Cloud!)

**Comment by yakkomajuri** [comment]
This is ready for another go - it mostly just moves the logic but probably could use some selectors etc

**Comment by mariusandra** [comment]
Hey! Finally had the time to review it. I made quite some changes and cleanup though. 

Playing with this now, I think we need some "secret" or "write-only" field type for plugins, so that this key wouldn't be shown all the time. As in, someone could go into "plugins" and read their boss's personal API key and do something with it. I had a need for it with the github star sync plugin for example.

**Pull request merged**